resources/manager/data.go

```diff
@@ -115,22 +111,17 @@ func (d *Data) fetchManager(ctx context.Context, updates chan update) {
 
 	for {
 		select {
+		case <-ctx.Done():
+			return
+
 		case <-ticker.C:
 			// Generate input ID?
+			// Send aggregated data at cycle tick
+			d.output <- d.cycleData
```

Comment:
demo of the problem https://go.dev/play/p/tPwhNxo9hha

Reply:
The curious behavior is when we [add a sleep of one millisecond prior to consuming the channel](https://go.dev/play/p/kLT2HyrlkKJ), as you can see we are getting the original values.

This happens because `time.Sleep` blocks the go routine and advances the retrieval of value prior to our changes after sending the data to the channel.

After researching and coding some examples, I came to a conclusion that channels are following the golang's data types logic (unless the channel type is a pointer).

This means that in the case of [basic types](https://golangbyexample.com/all-data-types-in-golang-with-examples/#Basic_Types) and [composite non-reference types](https://golangbyexample.com/all-data-types-in-golang-with-examples/#Non-Reference_Types) channels will actually copy the whole input data.

But in the case of [composite reference types](https://golangbyexample.com/all-data-types-in-golang-with-examples/#Reference_Types) channels will actually hold a reference pointer and won't copy the input data.

You can look that it actually behaves like that by looking at the [same example](https://go.dev/play/p/ZAAhxXkk68g), but with the usage of an array instead of a slice.

The channel runs this logic recursively on the input data as you can see in the [following example](https://go.dev/play/p/r4zlH7cOh9i), I've picked some types to run with in order to visualize the behavior.

The cycleData is a map of slices (`type ResourceMap map[string][]fetching.Resource`), which means that it is being passed with a reference pointer to the channel.

There are several ways of how we can handle this problem:

1. Refactor and use types that are being copied (still leaves us without control over the SDK structures, but I assume that they won't be re-used while the data is in the channel), that way we will be able to pass it to the channel and modify right afterward without any concerns.

2. Once passing the data to the channel, allocating new slot for upcoming modifications while the channel holds the previous reference and the data will get garbage collected when it will be read from the channel and handled automatically ([see the following example](https://go.dev/play/p/Mt5_b9bfvaZ)).

There are probably more options, but these are the ones that crossed my mind.
I think that option 2 is probably the easiest and quickest to implement.
I shall also note that this will be a temporary solution until we refactor the whole flow (which will remove the cycleData).